odoo/models.py

```diff
@@ -1228,8 +1228,16 @@ def _log(base, record, field, exception):
                 try:
                     dbid = int(record['.id'])
                 except ValueError:
-                    # in case of overridden id column
-                    dbid = record['.id']
+                    if self._fields["id"].type != "integer":
+                        # in case of overridden id column
+                        dbid = record['.id']
+                    else:
+                        log(dict(extras,
+                            type='error',
+                            record=stream.index,
+                            field='.id',
+                            message=_(u"Invalid database identifier '%s'") % dbid))
+
                 if not self.search([('id', '=', dbid)]):
```

Comment:
@rco-odoo what do you think about modifying the search here and adding a with_context(active_test=False) ?
Indeed if you want to import (using id) in order to active a archived product, the search will not find the id